### PR TITLE
Devel/l.bryndza/unit tests qemu

### PR DESCRIPTION
**Pull request by lucckb**

### CODE REVIEW

File: extras/doc/unit_test_qemu.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### Compile QEMU
+
+To compile QEMU first download the patch from the following location: http://bryndza.boff.pl/downloads/prv/0001-STM32-fix-raise-interrupt-time.patch and then clone the QEMU sources and apply the patch and compile the application.
+
+```bash
+git clone https://gitlab.com/qemu-project/qemu.git
+cd qemu
+git submodule init
+git submodule update --recursive
```
Comment (tomeksowi) [inline extras/doc/unit_test_qemu.md]:
```suggestion
git clone --recurse-submodules https://gitlab.com/qemu-project/qemu.git
cd qemu
```

File: tests/libisix/03_mutex.cpp [hunk 2]
```diff
@@ -421,8 +424,8 @@ static const lest::test module[] =
 		isix::wait_ms(20);
 		EXPECT( isix::mutex_destroy( mloc )==ISIX_EOK );
 		isix::wait_ms(10);
-		EXPECT( retp[0]==ISIX_EOK );
-		EXPECT( retp[1]==ISIX_EDESTROY );
+		EXPECT( retp[0]==ISIX_EOK );	// First obtained
+		EXPECT( retp[1]==ISIX_EDESTROY );	// Others wating and destroyed
```
Comment (tomeksowi) [inline tests/libisix/03_mutex.cpp]:
```suggestion
		EXPECT( retp[1]==ISIX_EDESTROY );	// Others waiting and destroyed
```

File: tests/libisix/07_vtimer.cpp [hunk 3]
```diff
@@ -178,7 +178,7 @@ static const lest::test module[] =
 		EXPECT( m_t1.stop_isr()==ISIX_EOK );
 		EXPECT( m_t2.stop_isr()==ISIX_EOK );
 		EXPECT( m_t3.stop_isr()==ISIX_EOK );
-		isix_wait_ms(50);	//Give some time to command exec
+		isix_wait_ms(t3+2);	//Give some time to command exec
```
Comment (tomeksowi) [inline tests/libisix/07_vtimer.cpp]:
```suggestion
		isix_wait_ms(t3+2);	//Give some time to exec command
```

File: tests/libisix/unittests_entry.cpp [hunk 4]
```diff
@@ -26,27 +26,83 @@
...
+                throw std::bad_alloc();
+            } catch (const std::exception& e) {
+				ec = 1;
+            } catch (...) {
+				ec = 2;
+            }
+			return ec;
+        };
+		if( basic_exc() != 1) {
+			dbprintf("Exceptions test failed. Unable to continue with least");
```
Comment (tomeksowi) [inline tests/libisix/unittests_entry.cpp]:
```suggestion
			dbprintf("Exceptions test failed. Unable to continue with lest");
```

File: tests/libisix/unittests_entry.cpp [hunk 5]
```diff
@@ -26,27 +26,83 @@
...
+				rethrow_exc();
+			} catch( std::exception& ex) {
+				ec = 1;
+			} catch(...) {
+				ec = 2;
+			}
+			return ec;
+		};
+		if(retrow_catch() != 1) {
+			dbprintf("Exceptions rethow tests failed. Unable to continue with least");
```
Comment (tomeksowi) [inline tests/libisix/unittests_entry.cpp]:
```suggestion
			dbprintf("Exceptions rethow tests failed. Unable to continue with lest");
```

File: tests/libisix/utils/board_boot.cpp [hunk 6]
```diff
@@ -63,6 +66,74 @@ namespace {
...
+			return false;
+		}
+#endif
+		LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_PLL);
+		for( auto r=0; r<retries; ++r ) {
+			if( LL_RCC_GetSysClkSource() == LL_RCC_SYS_CLKSOURCE_STATUS_PLL ) {
+				break;
+			}
+		}
+        // Ignore quemu
```
Comment (tomeksowi) [inline tests/libisix/utils/board_boot.cpp]:
```suggestion
        // Ignore qemu
```

File: libperiph/vendor/arm/cortexm/stm32/f4/src/stm32f4xx_ll_rcc.c [hunk 7]
```diff
@@ -765,7 +765,7 @@ uint32_t LL_RCC_GetSDIOClockFreq(uint32_t SDIOxSource)
   * @retval RNG clock frequency (in Hz)
   *         - @ref  LL_RCC_PERIPH_FREQUENCY_NO indicates that oscillator is not ready
   */
-uint32_t LL_RCC_GetRNGClockFreq(uint32_t RNGxSource)
+uint32_t LL_RCC_GetRNGClockFreq(uint32_t __attribute__((unused)) RNGxSource)
```
Comment (tomeksowi) [inline libperiph/vendor/arm/cortexm/stm32/f4/src/stm32f4xx_ll_rcc.c:768]: When is RNGxSource unused?

**Review by tomeksowi — CHANGES_REQUESTED** [review]
LGTM, small suggestions

File: libperiph/vendor/arm/cortexm/stm32/f4/src/stm32f4xx_ll_rcc.c (hunk 7 above)
Comment (lucckb) [reply libperiph/vendor/arm/cortexm/stm32/f4/src/stm32f4xx_ll_rcc.c:768]: for some microcontrollers that are selected by the ifdef directive the parameter is not used which generates a warning

**Review by tomeksowi — APPROVED** [review]

**Pull request merged**